script/sign_upload_linux_pkgs.py

```diff
@@ -0,0 +1,243 @@
+                logging.info("RPM signing successful!")
+            except subprocess.CalledProcessError as cpe:
+                logging.error("Error running command: \"{}\"".format(log_cmd))
+                exit(1)
+
+    try:
+        os.chdir(repo_dir)
+        logging.debug('Changed directory to \"{}\"'.format(repo_dir))
+    except OSError as ose:
+        message = ('Error: could not change directory to {}: {}'.format(repo_dir, ose))
+        exit(message)
+
+    # Now upload to aptly and rpm repos
+
+    for item in ['upload_to_aptly', 'upload_to_rpm_repo']:
+        bucket = ''
+        if re.match(r'.*rpm.*', item):
+            bucket = 'brave-browser-rpm-staging-'
+        else:
+            bucket = 'brave-browser-apt-staging-'
+
+        upload_script = os.path.join(repo_dir, item)
+
+        TESTCHANNEL = 'test'
+
+        if s3_test_buckets:
+            upload_cmd = '{} {} {}'.format(upload_script, bucket + channel + '-' +
+                                           TESTCHANNEL, gpg_fingerprint)
+        else:
+            upload_cmd = '{} {} {}'.format(upload_script, bucket + channel, gpg_fingerprint)
+        logging.info("Running command: \"{}\"".format(upload_cmd))
+        try:
+            subprocess.check_output(upload_cmd, shell=True)
+        except subprocess.CalledProcessError as cpe:
+            logging.error("Error: {}".format(cpe))
+            exit(1)
+
+    # Not sure we need to change back to the original dir here,
+    # keeping it for now.
+    try:
+        os.chdir(saved_path)
+        logging.debug('Changed directory to \"{}\"'.format(saved_path))
+    except OSError as ose:
+        message = ('Error: could not change directory to {}: {}'.format(saved_path, ose))
+        exit(message)
+
+
+def download_from_github(args, logging):
+
+    file_list = []
+
+    # BRAVE_REPO defined in helpers.py
+    repo = GitHub(args.github_token).repos(BRAVE_REPO)
+    tag_name = args.tag
+    release = {}
+    releases = get_releases_by_tag(repo, tag_name, include_drafts=True)
+    if releases:
+        if len(releases) > 1:
+            exit("Error: More than 1 release exists with the tag: \'{}\'".format(tag_name))
+        release = releases[0]
+        logging.info("Searching for RPM/DEB packages in GitHub release: {}".format(release['url']))
+    for asset in release['assets']:
+        if re.match(r'.*\.rpm$', asset['name']) \
+                or re.match(r'.*\.deb$', asset['name']):
+            filename = asset['name']
+            asset_id = asset['id']
+            asset_url = asset['url']
+            if args.debug:
+                logging.debug("GitHub asset_url: {}".format(
+                    asset_url + '/' + filename))
+
+            # Check if the file exists on disk first, and rename it if so
+            # to prevent a situation where the expect script fails because
+            # the rpm has already been signed
+            if os.path.isfile(os.path.join(os.getcwd(), filename)):
```

Comment:
for the purpose of unit testing, it would be great if this block (and other blocks, like the download block below) were broken out to a function. ex: `rename_file_if_exists`

Reply:
> has this been flagged for security review?

I will open a security review now @bridiver.

Reply:
Moved a couple parts of that larger method into their own methods.